### ISSUE TITLE
Add Support for Maps

### ISSUE THREAD

**Issue by fionera** [go-ini/ini#197]
### Please give general description of the problem
In the PHP implementation of ini loading you can define maps. This would be a good addition for the parser.
 https://php.net/manual/en/function.parse-ini-file.php

### Do you have any suggestion to fix the problem?

**Comment by Unknwon** [comment]
Hi,
>Please provide code snippets to reproduce the problem described above

**Comment by fionera** [comment]
Something like this? 

ini:
```
[section]
phpversion[] = "5.0"
phpversion[] = "5.1"
phpversion[] = "5.2"
phpversion[] = "5.3"

urls[svn] = "http://svn.php.net"
urls[git] = "http://git.php.net"
```

go:
```
var phpVersions []string
phpVersions = file.Section("section").Key("phpversion").Array()

var urls map[string]string
urls = file.Section("section").Key("urls").StringMap()
```

**Comment by Unknwon** [comment]
Thanks for the follow up!

This library has [Shadows keys](https://ini.unknwon.io/docs/howto/work_with_keys#same-key-with-multiple-values) for the case of `phpversion[]`. For the case of `urls[svn]`, currently not supported.

**Comment by fionera** [comment]
Yes I know :) Is it hard to add support for it? Would be a very nice feature ^^

**Comment by unknwon** [comment]
It would be cool to have, but feels more like a PHP-specific feature?

**Comment by fionera** [comment]
I changed the config backend in rclone to use this... In my case I need it to save a structure like this:

```
path["test"] = test_remote:
path["test/foo"] = blafoo:
```

**Comment by unknwon** [comment]
Hi,

```
path["test"] = test_remote:
path["test/foo"] = blafoo:
```

For this, do you want read in from somewhere else or you are trying write/save in this format?

**Comment by fionera** [comment]
I need to read/write in that format because I cant change the way the other config is working

**Comment by unknwon** [comment]
Got it. It is a cool feature request, but currently do not bandwidth to work on this.

**Comment by TomWright** [comment]
I would also appreciate this feature.

I would like to parse
```
a=1

[section1]
b=2
```

Into 
```
map[string]interface{}{
	"a": 1,
	"section1": map[string]interface{}{
		"b": 2,
	},
}
```

**Comment by unknwon** [comment]
@TomWright Hi, based on your example, what you're asking is a completely different thing from the original request :)

**Comment by TomWright** [comment]
@unknwon I thought the original request was asking that we could decode into a generic map type. Sorry if I've misunderstood 🙂

If that is the case, do you know if what I ask is possible? I'd also need to go from a `map[string]interface{}` into the ini format. I'm happy to raise another issue with more precise expectations of that would help.

**Comment by unknwon** [comment]
Yes, please file another issue :)
